recipes/cppkafka/all/conanfile.py

```diff
@@ -0,0 +1,134 @@
+       "shared": [True, False],
+       "fPIC": [True, False],
+       "multithreaded": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "multithreaded": True,
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        self.options["librdkafka"].shared = self.options.shared
+
+    def requirements(self):
+        self.requires("boost/1.78.0")
+        self.requires("librdkafka/1.8.2")
+
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "6",
+            "clang": "4",
+            "apple-clang": "5.1",
+        }
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 14)
+
+        minimum_version = self._minimum_compilers_version.get(str(self.settings.compiler), False)
+        if not minimum_version:
+            self.output.warn(
+                "{}/{} requires C++14. Your compiler is unknown. Assuming it supports C++14.".format(self.name, self.version,))
+        elif tools.Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                "{}/{} requires C++14, which your compiler does not support.".format(self.name, self.version,))
+
+        if self.settings.compiler in ["clang", "apple-clang"] and self.settings.compiler.get_safe("libcxx") == "libc++":
+            raise ConanInvalidConfiguration(
+                "{}/{} doesn't support {} with libc++".format(self.name, self.version, self.settings.compiler))
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["CPPKAFKA_BUILD_SHARED"] = self.options.shared
+        cmake.definitions["CPPKAFKA_BOOST_USE_MULTITHREADED"] = self.options.multithreaded
+        cmake.definitions["CPPKAFKA_RDKAFKA_STATIC_LIB"] = not self.deps_cpp_info["librdkafka"].shared
+
+        cxx_flags = list()
+        if self.settings.get_safe("compiler.libcxx") == "libstdc++":
+            cxx_flags.append("-D_GLIBCXX_USE_CXX11_ABI=0")
+        elif self.settings.get_safe("compiler.libcxx") == "libstdc++11":
+            cxx_flags.append("-D_GLIBCXX_USE_CXX11_ABI=1")
+        if not self.options.shared:
+            cxx_flags.append("-DCPPKAFKA_STATIC")
```

Comment:
useless:
- `_GLIBCXX_USE_CXX11_ABI` is injected by conan
- `CPPKAFKA_STATIC` is defined upstream when `CPPKAFKA_BUILD_SHARED` is OFF: https://github.com/mfontanini/cppkafka/blob/0.4.0/CMakeLists.txt#L82-L91